tools/quadrados/src/entry.cpp

```diff
@@ -4,32 +4,89 @@
 #include "tools.hpp"
 
 /// Prints the help message of the program.
-int runHelp(int /*argc*/, char** /*argv*/)
+int runHelp(int /*argc*/, char** /*argv*/, GlobalArgs& /*ga*/)
 {
-    std::cerr << "Usage: quadrados <TOOL>" << std::endl;
+    std::cerr << "Usage: quadrados [OPTIONS] <TOOL>" << std::endl;
     std::cerr << "Tools:" << std::endl;
     for (const auto& tool : Tools)
     {
         std::cerr << "  " << tool.name << std::endl;
     }
+    std::cerr << "Options:" << std::endl;
+    std::cerr << "  -D <install_directory> path to install directory" << std::endl;
+
     return 0;
 }
 
+/*
+/   Checks if a string is a tool
+*/
+bool isTool(char* param)
+{
+    for (const auto& tool : Tools)
+    {
+        if (tool.name == param)
+        {
+
+            return true;
+        }
+    }
+
+    return false;
+}
+
+/*
+/   Returns the number of parsed args
+/   and updates the GlobalArgs struct if needed
+*/
+int parseGlobalArgs(int argc, char** argv, GlobalArgs& ga)
+{
+    int nParsed = 0;
+    for (int i = 1; i < argc; i++)
+    {
+        std::string arg = argv[i];
+        if (arg == "-D" && i + 1 < argc)
+        {
+            ga.installDir = argv[++i];
+            nParsed += 2;
+        }
+        else if (arg == "-D")
+        {
+            return -1;
+        }
+        else if (isTool(argv[i]))
+        {
+            return nParsed;
+        }
+    }
+
+    return -1;
+}
+
 int main(int argc, char** argv)
 {
-    // Parse command line arguments.
-    if (argc >= 2)
+    // Parse global args
+    GlobalArgs globalArgs;
```

Comment:
```suggestion
    GlobalArgs globalArgs;
    globalArgs.installDir = INSTALL_PATH;
```
Instead of having that logic in `getTemplateDir`, why not simply set the default value of the option to the macro here?
Then, in the tools, you can just use `globalArgs.installDir + "/templates"` without any other logic.

Reply:
Não tenho a certeza se percebi bem.
Faço a mesma verificação, mas depois de dar o parse dos argumentos globais. e se estiver em ambiente de desenvolvimento os installDir fica o caminho na macro QUADRADOS_DEV_TEMPLATE_DIR?

Reply:
Não percebi a parte de fazer verificação, nao precisas de fazer nenhuma, certo? Mas sim, o valor default da opcao fica o da macro, é exatamente como os assets. Nesse caso probably e melhor mudar o nome para INSTALL_PATH é ou assim porque nao é so para dev - quando estas em dist tambem e utilizada, simplesmente fica um caminho relativo